### PR TITLE
Rover add fence support

### PR DESCRIPTION
**Pull request by khancyr**
Add fence support without adding fence to ap_arming.
Need InertialNav removal !

### CODE REVIEW

**Comment by khancyr** [comment]
Rebased and correct this error and another one from rebasing.

File: APMrover2/system.cpp [hunk 1]
```diff
@@ -386,3 +393,70 @@ bool Rover::disarm_motors(void)
...
+// ekf_position_ok - returns true if the ekf claims it's horizontal absolute position estimate is ok and home position is set
+bool Rover::ekf_position_ok()
+{
+    if (!ahrs.have_inertial_nav()) {
+        // do not allow navigation with dcm position
+        return false;
+    }
+
+    // with EKF use filter status and ekf check
+    nav_filter_status filt_status = inertial_nav.get_filter_status();
```
Comment (amilcarlucas) [inline APMrover2/system.cpp]: This requires inertial_nav. And Inertial Nav has been removed in https://github.com/ArduPilot/ardupilot/pull/6720

File: APMrover2/system.cpp [hunk 2]
```diff
@@ -386,3 +393,70 @@ bool Rover::disarm_motors(void)
...
+    }
+#endif
+#if VISUAL_ODOMETRY_ENABLED == ENABLED
+    if (!g2.visual_odom.enabled()) {
+        return false;
+    }
+#endif
+
+    // get filter status from EKF
+    nav_filter_status filt_status = inertial_nav.get_filter_status();
```
Comment (amilcarlucas) [inline APMrover2/system.cpp]: This requires inertial_nav. And Inertial Nav has been removed in https://github.com/ArduPilot/ardupilot/pull/6720

**Comment by khancyr** [comment]
Rebased on https://github.com/ArduPilot/ardupilot/pull/7377

**Comment by khancyr** [comment]
merge by Randy

**Pull request closed without merging**